plugins/color/gsd-color-state.c

```diff
@@ -46,6 +47,70 @@
 
 static void gcm_session_set_gamma_for_all_devices (GsdColorState *state);
 
+static guint gcm_session_get_output_percentage (void);
+
+typedef struct {
+  unsigned short vendor;
+  unsigned short product;
+} EdidInfo;
+
+typedef struct {
+  unsigned long pattern;
+  EdidInfo info;
+} EdidHeader;
+
+static EdidInfo
+known_oled[] = {
+  // SDC OLED panel in S76 Adder and XPS 7590
+  { 19587, 41001 }
+};
+
+static gboolean
+has_oled_builtin_display (void)
+{
+  gboolean has_oled = FALSE;
+
+  // TODO: The Rust code does a dir walk to find all outputs. Needed?
+  FILE *f = fopen("/sys/class/drm/card0-eDP-1/edid", "rb");
```

Comment:
This is okay for now, if both the supported models end up having the OLED output at the same path. But I would be concerned if the detection orders of GPUs is not deterministic.

Reply:
Would the long-term solution be to interact with the EDID data being passed around elsewhere in the file?